Linux binaries fail to compile in GitHub releases after #76266

### Describe the bug

Linux Binaries are not generating after #76266

### Attach save file

N/A

### Steps to reproduce

N/A

### Expected behavior

I expect all standard binaries to generate after a code update.

Perhaps it would be a good idea for GitHub to run a test to see how many files are generated after a commit.

### Screenshots

![image](https://github.com/user-attachments/assets/b6fe2b59-9074-4ab4-b952-7560df2908ca)


### Versions and configuration

N/A

### Additional context

_No response_

There is a Problem at the Build - GCC 9, Curses, LTO -  get an Error, and so the Build for Linux (ubuntu) get cancelled.
The Later Builds for Linux were all cancelled. I have seen in the newest Build Progress, that there is an Error.
(Error: src/mtype.h:320:19: error: declaration of ‘baby_type mtype::baby_type’ changes meaning of ‘baby_type’ [-fpermissive])

--- a/src/mtype.h
+++ b/src/mtype.h
@@ -280,10 +280,10 @@ struct mount_item_data {
 };
 
 struct reproduction_type {
-    mtype_id baby_monster;
-    mongroup_id baby_monster_group;
-    itype_id baby_egg;
-    item_group_id baby_egg_group;
+    mtype_id baby_monster = mtype_id::NULL_ID();
+    mongroup_id baby_monster_group = mongroup_id::NULL_ID();
+    itype_id baby_egg = itype_id::NULL_ID();
+    item_group_id baby_egg_group = item_group_id::NULL_ID();
 };
 
 struct mtype {

--- a/src/mtype.h
+++ b/src/mtype.h
@@ -279,7 +279,7 @@ struct mount_item_data {
     itype_id storage;
 };
 
-struct baby_type {
+struct reproduction_type {
     mtype_id baby_monster;
     mongroup_id baby_monster_group;
     itype_id baby_egg;
@@ -317,7 +317,7 @@ struct mtype {
         mtype_id zombify_into; // mtype_id this monster zombifies into
         mtype_id fungalize_into; // mtype_id this monster fungalize into
 
-        baby_type baby_type;
+        reproduction_type baby_type;
 
         // Monster biosignature variables
         itype_id biosig_item;

